aedile-core/src/main/kotlin/com/sksamuel/aedile/core/caffeine.kt

```diff
@@ -125,6 +132,10 @@ fun <K, V> caffeineBuilder(configure: Configuration<K, V>.() -> Unit = {}): Buil
 
    val scope = c.scope ?: CoroutineScope(c.dispatcher + CoroutineName("Aedile-Caffeine-Scope") + SupervisorJob())
 
+   val scopeProvider: suspend () -> CoroutineScope = {
+      if (c.useCallingContext) CoroutineScope(coroutineContext) else scope
```

Comment:
This change here is the one I wasn't sure about - whether we should append or create a new scope, as we lose the dispatcher / supervisor job, I suppose we can create it with them every time if that's what you mean?